crates/supervisor/storage/src/providers/log_provider.rs

```diff
@@ -124,6 +124,19 @@ where
         }
         Ok(())
     }
+
+    pub(crate) fn rewind_from(&self, block_number: u64) -> Result<(), StorageError> {
```

Comment:
I guess `rewind_to` name makes more sense considering we are dealing with the append only logging.